src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts

```diff
@@ -138,41 +172,60 @@ const buildListSectionsData = ({
     );
   }
 
+  if (combinedData.popularAssets?.length) {
+    const filteredPopular = filterAssetsFromBridgeAndRecent({
+      assets: combinedData.popularAssets,
+      recentSwaps: combinedData.recentSwaps,
+      filteredBridgeAssetAddress,
+    }).slice(0, 6);
```

Comment:
If we're going to limit the popular tokens to the top 6, we should add a selector to the react-query instead of hiding it here.

Reply:
i need to slice the first 6 AFTER this filtering is done though. if the slicing is done at the query level then it could get filtered down to less than 6 items.

Reply:
oh true fair point. This brings up an interesting point that maybe we need to move the filtering to react-query itself.